glib/src/subclass/object.rs

```diff
@@ -231,47 +232,31 @@ mod test {
         use super::*;
 
         // A dummy `Object` to test setting an `Object` property and returning an `Object` in signals
+        #[derive(Default)]
         pub struct ChildObject;
+
+        #[glib::object_subclass]
         impl ObjectSubclass for ChildObject {
             const NAME: &'static str = "ChildObject";
             type Type = super::ChildObject;
             type ParentType = Object;
-            type Interfaces = ();
-            type Instance = subclass::simple::InstanceStruct<Self>;
-            type Class = subclass::simple::ClassStruct<Self>;
-
-            object_subclass!();
-
-            fn new() -> Self {
-                ChildObject
-            }
         }
 
         impl ObjectImpl for ChildObject {}
 
+        #[derive(Default)]
         pub struct SimpleObject {
             name: RefCell<Option<String>>,
             construct_name: RefCell<Option<String>>,
             constructed: RefCell<bool>,
         }
 
+        #[glib::object_subclass]
         impl ObjectSubclass for SimpleObject {
             const NAME: &'static str = "SimpleObject";
             type Type = super::SimpleObject;
             type ParentType = Object;
             type Interfaces = (DummyInterface,);
```

Comment:
If we are friendly we could also allow for a single interface either
  - Passing it in without the trailing comma and add one then
  - Give a more friendly error if the trailing comma is missing

Reply:
Perhaps it could offer a more friendly error message. Though I'm inclined not to add any special handling here. Personally I don't really mind how it's specified now since I'm familiar with how single element tuples are written, and I like that everything the macro is currently doing might ideally be a feature that a future Rust version could natively support.

Reply:
Yeah I agree. A more friendly error then maybe, but that can also happen in a future PR if it's not trivial to add.